pymc3/backends/arviz.py

```diff
@@ -0,0 +1,676 @@
+                None
+                if var_names is None
+                else dict_to_dataset(
+                    {k: np.expand_dims(self.prior[k], 0) for k in var_names},
+                    library=pymc3,
+                    coords=self.coords,
+                    dims=self.dims,
+                    # index_origin=self.index_origin,
+                )
+            )
+        return priors_dict
+
+    @requires(["observations", "multi_observations"])
+    @requires("model")
+    def observed_data_to_xarray(self):
+        """Convert observed data to xarray."""
+        if self.predictions:
+            return None
+        return dict_to_dataset(
+            {**self.observations, **self.multi_observations},
+            library=pymc3,
+            coords=self.coords,
+            dims=self.dims,
+            # default_dims=[],
+            # index_origin=self.index_origin,
+        )
+
+    @requires(["trace", "predictions"])
+    @requires("model")
+    def constant_data_to_xarray(self):
+        """Convert constant data to xarray."""
+        # For constant data, we are concerned only with deterministics and data.
+        # The constant data vars must be either pm.Data (TensorSharedVariable) or pm.Deterministic
+        constant_data_vars = {}  # type: Dict[str, Var]
+        for var in self.model.deterministics:
+            ancestor_vars = ancestors(var.owner.inputs)
+            # no dependency on a random variable
+            if not any(isinstance(a, TensorVariable) for a in ancestor_vars):
+                constant_data_vars[var.name] = var
+
+        def is_data(name, var) -> bool:
+            assert self.model is not None
+            return (
+                var not in self.model.deterministics
+                and var not in self.model.observed_RVs
+                and var not in self.model.free_RVs
+                and var not in self.model.potentials
+                and (self.observations is None or name not in self.observations)
+            )
+
+        # I don't know how to find pm.Data, except that they are named variables that aren't
+        # observed or free RVs, nor are they deterministics, and then we eliminate observations.
+        for name, var in self.model.named_vars.items():
+            if is_data(name, var):
+                constant_data_vars[name] = var
+
+        if not constant_data_vars:
+            return None
+        constant_data = {}
+        for name, vals in constant_data_vars.items():
+            if hasattr(vals, "get_value"):
+                vals = vals.get_value()
+            # this might be a Deterministic, and must be evaluated
+            elif hasattr(self.model[name], "eval"):
+                vals = self.model[name].eval()
+            constant_data[name] = vals
+        return dict_to_dataset(
+            constant_data,
+            library=pymc3,
+            coords=self.coords,
+            dims=self.dims,
+            # default_dims=[],
```

Comment:
This won't work yet I think, `dict_to_dataset` will try to add the `chain` and `draw` dimensions and fail. The old version (with old meaning the current release) had all the code in `dict_to_dataset` repeated here to not assume `chain` and `draw` were present.

Reply:
`dims=self.dims` won't work?

Reply:
no, commenting `default_dims=[]` because it should revert to the default behaviour of assuming `chain` and `draw` are present and they are not.

Reply:
`test_idata_conversion.py` passes locally with that commented out, but I had to comment out `dims` as well.